### ISSUE TITLE
Python code for scRNA analysis

### ISSUE THREAD

**Issue by jlause** [ZOC-Liu-Lab/RGC-Patchseq#1]
Dear all,

very interesting paper! I wanted to learn a bit more about your t-type mapping, so I was looking for the code you used to assign the cell type labels from Tran et al. 2019 to your patchSeq RNA data. Your paper mentions that you use a sklearn implementation and Scanpy to do that in python. I could not find it in this repo, so could you add it here or point me to where I can find it?

Thank you very much!
Kind regards,
Jan

**Comment by ZOC-Liu-Lab** [comment]
Dear Jan, 


Thank you for your interest in our paper! Sorry for the delay.


Has any one replied to you yet? 


If not, please let me know. I will see what I can do for you.


Best,


Zhaozhe




At 2022-09-22 05:46:02, "Jan Lause" ***@***.***> wrote:

Dear all,

very interesting paper! I wanted to learn a bit more about your t-type mapping, so I was looking for the code you used to assign the cell type labels from Tran et al. 2019 to your patchSeq RNA data. Is it in this repo or if not, could you add it here or Point be to where I can find it?

Thank you very much!
Kind regards,
Jan

**Comment by jlause** [comment]
Dear Zhaozhe,
Thanks for your reply! I have not heard from anyone else yet and would still be interested in the details/code of your t-type mapping - looking forward to learn more about it!
Best, Jan

**Comment by ZOC-Liu-Lab** [comment]
> Dear Zhaozhe, Thanks for your reply! I have not heard from anyone else yet and would still be interested in the details/code of your t-type mapping - looking forward to learn more about it! Best, Jan

Dear Jan,

I have uploaded the code we used for t-type mapping. Briefly, we used a logistic-regression based machine learning algorithm to predict the patch-seq label, based on the gene expression pattern of the 10x data.  (The function label_projection)

To test the consistency of the results, we went through a range of parameters during the training (and also the PCA and harmony-corrected PCA space too). 

Please feel free to let me know if you need further assistance.

Best,

Zhaozhe

**Comment by jlause** [comment]
Dear Zhaozhe,
thanks for the upload, very interesting!
I have trouble finding the following functions:

- `data_subset()`
- `sele_dataset()`
- `run_PCA_Harmony()`
- `train_test_split_per_type()`
- `find_save_score()`

I checked the [SCCAF repo](https://github.com/SCCAF/sccaf), but could not find them there. Are these also custom functions? Could you add them here as well?

Thanks a lot  and looking forward!
Best, Jan

**Comment by ZOC-Liu-Lab** [comment]
Thank you for the feedback, Jan!



Sorry about the inconvinience. I have added the function at the top of the (updated) script. 


Please feel free to let me know if there is further functions missing. 


Zhaozhe







At 2022-10-21 23:57:30, "Jan Lause" ***@***.***> wrote:

Dear Zhaozhe,
thanks for the upload, very interesting!
I have trouble finding the following functions:

data_subset()
sele_dataset()
run_PCA_Harmony()
train_test_split_per_type()
find_save_score()

I checked the SCCAF repo, but could not find them there. Are these also custom functions? Could you add them here as well?

Thanks a lot and looking forward!
Best, Jan
